Error forwarding `compile` commands when devcontainer has no metadata

Hi there!

When I try to forward any `compile` command using `devcontainer-mode` to basic devcontainers like this one:

```json
{
  "name": "toy devcontainer",
  "image": "some_image"
}
```
I get the following error: `End of file while parsing JSON: 1, 0, 0`. 

Investing it I found that the cause was in `devcontainer--container-metadata`. This function parses the string returned from inspecting the container, but sometimes the container has no metadata at all and the inspection returns an empty string that fails to be parsed as JSON.

Thanks for reporting.

I cannot reproduce your problem.

A project only consisting of `.devcontainer/devcontainer.json` with the following content

```json
{
    "name": "toy devcontainer",
    "image": "mcr.microsoft.com/vscode/devcontainers/base:alpine"
}
```
seems to work.

`compile` with the default `make -k` gives

```
-*- mode: compilation; default-directory: "/tmp/testproject/" -*-
Compilation started at Fri Dec  5 19:27:12

docker exec --workdir /workspaces/testproject/ --user vscode 1200711b9afb make -k 
make: *** No targets specified and no makefile found.  Stop.

Compilation exited abnormally with code 2 at Fri Dec  5 19:27:13, duration 0.07 s
```

 Maybe you can MRE.

--- a/devcontainer.el
+++ b/devcontainer.el
@@ -587,16 +587,17 @@ If IN-TERMINAL is non-nil, also the ones of
 
 If IN-TERMINAL is non nil, the \"-it\" flag is set."
   (when-let* ((container-id (devcontainer-up-container-id)))
-    (remq nil
-          (append
-           (list
-            (symbol-name devcontainer-engine)
-            "exec"
-            (when in-terminal "-it")
-            "--workdir" (devcontainer-remote-workdir)
-            "--user" (devcontainer-remote-user))
-           (devcontainer--make-env-cli-args in-terminal)
-           (list container-id)))))
+    (let ((remote-user (devcontainer-remote-user)))
+      (remq nil
+            (append
+             (list
+              (symbol-name devcontainer-engine)
+              "exec"
+              (when in-terminal "-it")
+              "--workdir" (devcontainer-remote-workdir)
+              (when remote-user "--user") remote-user)
+             (devcontainer--make-env-cli-args in-terminal)
+             (list container-id))))))
 
 (defun devcontainer--fix-quoted-env-elements (command-string)
   "Fix overquoted environment elements in COMMAND-STRING.

--- a/test/devcontainer.el-test.el
+++ b/test/devcontainer.el-test.el
@@ -682,6 +682,19 @@
          (should (equal (devcontainer-advise-command "grep foo") cmd))
          (devcontainer--compile-start-advice #'my-compile-fun "grep foo"))))))
 
+(ert-deftest compilation-start-no-user ()
+  (devcontainer-mode 1)
+  (fixture-tmp-dir "test-repo-devcontainer"
+    (let ((cmd "docker exec --workdir /workspaces/project/ abcdef grep foo")
+          (devcontainer-execute-outside-container nil))
+      (mocker-let ((my-compile-fun (command &rest rest) ((:input `(,cmd))))
+                   (devcontainer-remote-workdir () ((:output "/workspaces/project/")))
+                   (devcontainer-remote-user () ((:output nil)))
+                   (devcontainer-remote-environment () ((:output nil)))
+                   (devcontainer-up-container-id () ((:output "abcdef"))))
+        (should (equal (devcontainer-advise-command "grep foo") cmd))
+        (devcontainer--compile-start-advice #'my-compile-fun "grep foo")))))
+
 (ert-deftest compilation-start-exclude-simple ()
   (devcontainer-mode 1)
   (fixture-tmp-dir "test-repo-devcontainer"

